redis_broadcast.go

```diff
@@ -355,9 +367,11 @@ func (bc *redisBroadcast) getNumSub(channel string) (int, error) {
 		return 0, err
 	}
 
-	var numSub64 int64
-	numSub64 = rs.([]interface{})[1].(int64)
-	return int(numSub64), nil
+	numSub64, ok := rs.([]interface{})[1].(int)
+	if ok {
+		return numSub64, nil
+	}
+	return 0, errors.New("redis reply cast to int error")
```

Comment:
A small suggestion to keep the happy path to the left (it's considered more idiomatic to keep the green path unindented)

```suggestion
	if !ok {
		return 0, errors.New("redis reply cast to int error")
	}
	return numSub64, nil
```

Reply:
i agree with you

Reply:
this one is guard code case, 

> This is an example of a common situation where code must guard against a sequence of error conditions. The code reads well if the successful flow of control runs down the page, eliminating error cases as they arise. Since error cases tend to end in return statements, the resulting code needs no else statements.
[effective_go](https://golang.org/doc/effective_go#if),error early return,

Reply:
Code at example, does not same in our example. It is different.

Reply:
Look up for this https://github.com/googollee/go-socket.io/blob/661adc4437cc7021270f1e6267ff9b3be845b9e4/redis_broadcast.go#L353
You chosen same way as recommended by Alexandros from next code

Reply:
I didn't got it,is your meaning that we should keep code consistency?
And cast error is also error,function should early return.
In this case i know there is no more codes follow after the cast expression,but if we would change the function in the futrue,we need do it too

Reply:
> Look up for this
> 
> https://github.com/googollee/go-socket.io/blob/661adc4437cc7021270f1e6267ff9b3be845b9e4/redis_broadcast.go#L353
> 
> 
> You chosen same way as recommended by Alexandros from next code

It's my bad,i will fixed it

Reply:
relax, it is normally :)